internal/distributed/proxy/httpserver/handler_v1.go

```diff
@@ -0,0 +1,582 @@
+		c.AbortWithStatusJSON(http.StatusOK, gin.H{HTTPReturnCode: response.Status.ErrorCode, HTTPReturnMessage: "check collections " + req.CollectionName + " exists fail", HTTPReturnError: response.Status.Reason})
+		return false, errors.New(response.Status.Reason)
+	} else {
+		return response.Value, nil
+	}
+}
+
+func (h *Handlers) RegisterRoutesToV1(router gin.IRouter) {
+	router.GET(VectorCollectionsPath, h.listCollections)
+	router.POST(VectorCollectionsCreatePath, h.createCollection)
+	router.GET(VectorCollectionsDescribePath, h.getCollectionDetails)
+	router.POST(VectorCollectionsDropPath, h.dropCollection)
+	router.POST(VectorQueryPath, h.query)
+	router.POST(VectorGetPath, h.get)
+	router.POST(VectorDeletePath, h.delete)
+	router.POST(VectorInsertPath, h.insert)
+	router.POST(VectorSearchPath, h.search)
+}
+
+func (h *Handlers) listCollections(c *gin.Context) {
+	dbName := c.DefaultQuery(HTTPDbName, DefaultDbName)
+	req := milvuspb.ShowCollectionsRequest{
+		DbName: dbName,
+	}
+	username, _ := c.Get(ContextUsername)
+	_, authErr := proxy.PrivilegeInterceptorWithUsername(c, username.(string), &req)
+	if authErr != nil {
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusUnauthorized, HTTPReturnMessage: authErr.Error()})
+		return
+	}
+	response, err := h.proxy.ShowCollections(c, &req)
+	if err != nil {
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "show collections fail", HTTPReturnError: err.Error()})
+	} else if response.Status.ErrorCode != commonpb.ErrorCode_Success {
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "show collections fail", HTTPReturnError: response.Status.Reason})
+	} else {
+		var collections []string
+		if response.CollectionNames != nil {
+			collections = response.CollectionNames
+		}
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusOK, HTTPReturnData: collections})
+	}
+}
+
+func (h *Handlers) createCollection(c *gin.Context) {
+	httpReq := CreateCollectionReq{
+		DbName:       DefaultDbName,
+		MetricType:   DefaultMetricType,
+		PrimaryField: DefaultPrimaryFieldName,
+		VectorField:  DefaultVectorFieldName,
+	}
+	if err := c.ShouldBindBodyWith(&httpReq, binding.JSON); err != nil {
+		c.AbortWithStatusJSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "check your parameters conform to the json format", HTTPReturnError: err.Error()})
+		return
+	}
+	if httpReq.CollectionName == "" || httpReq.Dimension == 0 {
+		c.AbortWithStatusJSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "collectionName and dimension are both required."})
+		return
+	}
+	has, err := h.hasCollection(c, httpReq.DbName, httpReq.CollectionName)
+	if err != nil {
+		return
+	}
+	if has {
+		c.AbortWithStatusJSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusOK, HTTPReturnMessage: "collection " + httpReq.CollectionName + " already exist."})
```

Comment:
There is no need for inspection in this place, as it is intended to trigger the idempotent logic on the RootCoord side.

Reply:
As i test on playground, it returned error if create the same collection repeatedly
https://docs.zilliz.com/reference/create-collection
![image](https://github.com/milvus-io/milvus/assets/135960789/db4287c1-b641-4c92-9be9-2866b2f228bd)

Reply:
@PowderLi The RESTful service on the Cloud side has not implemented interception.

Reply:
@PowderLi So you don't have to check if the list exists here.
@lentitude2tk